src/HDF5Msg.chpl

```diff
@@ -1155,6 +1082,129 @@ module HDF5Msg {
         }
     }
 
+    proc categorical_tohdfMsg(msgArgs: borrowed MessageArgs, st: borrowed SymTab) throws {
+        use C_HDF5.HDF5_WAR;
+        var mode: int = msgArgs.get("write_mode").getIntValue();
+
+        var filename: string = msgArgs.getValueOf("filename");
+        var file_format = msgArgs.get("file_format").getIntValue();
+        var group = msgArgs.getValueOf("dset");
+        const objType = msgArgs.getValueOf("objType"); // needed for metadata
+
+        var overwrite: bool = if msgArgs.contains("overwrite")
+                                then msgArgs.get("overwrite").getBoolValue()
+                                else false;
+
+        // access entries - types are currently always the same for each
+        var codes_entry = st.lookup(msgArgs.getValueOf("codes"));
+        var codes = toSymEntry(toGenSymEntry(codes_entry), int);
+        var cat_entry:SegStringSymEntry = toSegStringSymEntry(st.lookup(msgArgs.getValueOf("categories")));
+        var cats = new SegString("", cat_entry);
+        var naCodes_entry = st.lookup(msgArgs.getValueOf("NA_codes"));
+        var naCodes = toSymEntry(toGenSymEntry(naCodes_entry), int);
+        var perm_seg_exist: bool = false;
+        if msgArgs.contains("permutation") && msgArgs.contains("segments") {
+            perm_seg_exist = true;
+        }
```

Comment:
same question of it's possible to only have one or the other

Reply:
Again, based on the init we should have both or neither so I maintained that idea through